cli/src/main/java/com/box/l10n/mojito/cli/command/GitBlameCommand.java

```diff
@@ -0,0 +1,298 @@
+import org.eclipse.jgit.blame.BlameResult;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.fusesource.jansi.Ansi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author jaurambault
+ */
+@Component
+@Scope("prototype")
+@Parameters(commandNames = {"git-blame", "gb"}, commandDescription = "Git blame")
+public class GitBlameCommand extends Command {
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(GitBlameCommand.class);
+
+    @Autowired
+    ConsoleWriter consoleWriter;
+
+    @Parameter(names = {Param.REPOSITORY_LONG, Param.REPOSITORY_SHORT}, arity = 1, required = true, description = Param.REPOSITORY_DESCRIPTION)
+    String repositoryParam;
+
+    @Parameter(names = {Param.SOURCE_DIRECTORY_LONG, Param.SOURCE_DIRECTORY_SHORT}, arity = 1, required = false, description = Param.SOURCE_DIRECTORY_DESCRIPTION)
+    String sourceDirectoryParam;
+
+    @Parameter(names = {Param.FILE_TYPE_LONG, Param.FILE_TYPE_SHORT}, arity = 1, required = false, description = Param.FILE_TYPE_DESCRIPTION,
+            converter = FileTypeConverter.class)
+    FileType fileType;
+
+    @Parameter(names = {Param.SOURCE_LOCALE_LONG, Param.SOURCE_LOCALE_SHORT}, arity = 1, required = false, description = Param.SOURCE_LOCALE_DESCRIPTION)
+    String sourceLocale;
+
+    @Parameter(names = {Param.SOURCE_REGEX_LONG, Param.SOURCE_REGEX_SHORT}, arity = 1, required = false, description = Param.SOURCE_REGEX_DESCRIPTION)
+    String sourcePathFilterRegex;
+
+    @Parameter(names = {"--extracted-prefix"}, arity = 1, required = false, description = "prefix for path of extracted files")
+    String extractedFilePrefix;
+
+    @Autowired
+    AssetClient assetClient;
+
+    @Autowired
+    RepositoryClient repositoryClient;
+
+    @Autowired
+    TextUnitWithUsageClient textUnitWithUsageClient;
+
+    @Autowired
+    CommandHelper commandHelper;
+
+    CommandDirectories commandDirectories;
+
+    org.eclipse.jgit.lib.Repository gitRepository;
+
+    @Override
+    public void execute() throws CommandException {
+
+        commandDirectories = new CommandDirectories(sourceDirectoryParam);
+
+        consoleWriter.newLine().a("Git blame for repository: ").fg(Ansi.Color.CYAN).a(repositoryParam).println(2);
+
+        Repository repository = commandHelper.findRepositoryByName(repositoryParam);
+        List<PollableTask> pollableTasks = new ArrayList<>();
+
+        List<GitBlameWithUsage> textUnitsToBlame = textUnitWithUsageClient.getTextUnitToBlame(repository.getId());
+
+        if (fileType.getSourceFileExtension().equals("pot")) {
```

Comment:
it is not the best to check the extension in that case, instead you can check the file type with something like `if (fileType instanceOf POFileType)`.

There will be other type that will behave like PO file so it would make sense to move that as an attribute in the FileType but it can be done in another PR